src/subdomains/generic/kyc/services/name-check.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable } from '@nestjs/common';
+import fs from 'fs';
+import readline from 'readline';
+import { DfxLogger } from 'src/shared/services/dfx-logger';
+import { UserData } from '../../user/models/user-data/user-data.entity';
+import { DilisenseData } from '../dto/dilisense-data.dto';
+import { RiskRate } from '../entities/kyc-log.entity';
+import { KycLogService } from './kyc-log.service';
+
+@Injectable()
+export class NameCheckService {
+  private readonly logger = new DfxLogger(NameCheckService);
+
+  private readonly sanctionListPath = 'src/subdomains/generic/kyc/data/dilisense_consolidated_data_file.jsonl';
+  private sanctionData: DilisenseData[] = [];
+
+  constructor(private kycLogService: KycLogService) {
+    this.reloadSanctionList();
+  }
```

Comment:
Call from `onModuleInit` and put `void` (class should implement `OnModuleInit` interface, see e.g. MonitoringService)